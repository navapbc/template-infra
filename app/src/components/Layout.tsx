import { ReactElement } from 'react'
import { useTranslations } from 'next-intl'
import styles from '../../styles/Layout.module.scss'

type Props = {
  page?: ReactElement
}

const Layout = ({ page }: Props) => {
  const t = useTranslations('Layout')

  return (
    <>
      <header className={styles.header}>
        <em>{t('header')}</em>
      </header>
      <main className={styles.container}>
        {page}
      </main>
      <footer className={styles.footer}>
        <em>{t('footer')}</em>
      </footer>
    </>
  )
}

export default Layout
