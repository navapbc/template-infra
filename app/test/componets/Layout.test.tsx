// test/pages/index.test.js
import { axe } from 'jest-axe'
import { screen } from '@testing-library/react'
import renderWithIntl from '../renderWithIntl'
import Layout from '../../src/components/Layout'

describe('Layout', () => {
  it('should render placeholder header text', () => {
    renderWithIntl(<Layout />)

    const header = screen.getByText(/Template Header/i)

    expect(header).toBeInTheDocument()
    expect(header).toMatchSnapshot()
  })

  it('should pass accessibility scan', async () => {
    const { container } = renderWithIntl(<Layout />)
    const results = await axe(container)

    expect(results).toHaveNoViolations()
  })
})
