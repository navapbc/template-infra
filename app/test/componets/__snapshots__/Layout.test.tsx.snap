// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout should render placeholder header text 1`] = `
<em>
  Template header
</em>
`;
